File: app/src/main/java/com/arnold/uagroceries/ui/OnboardingAdapter.java
package com.arnold.uagroceries.ui;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

public class OnboardingAdapter extends FragmentStateAdapter {

    final static int NUM_PAGES = 3;


    public OnboardingAdapter(@NonNull FragmentActivity fragmentActivity) {
        super(fragmentActivity);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {

        if (position == 0) {
            return new Delivery1Fragment();

        } else if (position == 1) {
            return new Delivery2Fragment();

        } else {
            return new Delivery3Fragment();
        }

    }

    @Override
    public int getItemCount() {
        return NUM_PAGES;
    }
}

File: app/src/main/java/com/arnold/uagroceries/adapters/OnboardingAdapter.kt
package com.arnold.uagroceries.adapters

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.arnold.uagroceries.ui.onboarding.fragments.Delivery1Fragment
import com.arnold.uagroceries.ui.onboarding.fragments.Delivery2Fragment
import com.arnold.uagroceries.ui.onboarding.fragments.Delivery3Fragment

class OnboardingAdapter(fragmentActivity: FragmentActivity) : FragmentStateAdapter(fragmentActivity) {


    companion object {
        private const val NUM_OF_PAGES: Int = 3
    }

    override fun getItemCount(): Int {
        return NUM_OF_PAGES
    }

    override fun createFragment(position: Int): Fragment {

        return when (position) {
            0 -> {
                Delivery1Fragment()
            }
            1 -> {
                Delivery2Fragment()
            }
            else -> {
                Delivery3Fragment()
            }
        }
    }
}